demos/asp-net-core/ITOps.ViewModelComposition/CompositionHandler.cs

```diff
@@ -11,45 +11,42 @@ public class CompositionHandler
     {
         public static async Task<(dynamic ViewModel, int StatusCode)> HandleGetRequest(HttpContext context)
         {
-            var vm = new DynamicViewModel(context);
-            var pending = new List<Task>();
+            var viewModel = new DynamicViewModel(context.GetRouteData(), context.Request.Query);
             var routeData = context.GetRouteData();
-            var interceptors = context.RequestServices.GetServices<IRouteInterceptor>();
+
+            // matching interceptors could be cached by URL
+            var interceptors = context.RequestServices.GetServices<IRouteInterceptor>()
+                .Where(interceptor => interceptor.Matches(context.GetRouteData(), HttpMethods.Get))
+                .ToList();
 
             try
             {
-                //matching interceptors could be cached by URL
-                var matchingInterceptors = interceptors
-                    .Where(interceptor => interceptor.Matches(context.GetRouteData(), HttpMethods.Get))
-                    .ToArray();
-
-                foreach (var subscriber in matchingInterceptors.OfType<ISubscribeToCompositionEvents>())
+                foreach (var subscriber in interceptors.OfType<ISubscribeToCompositionEvents>())
```

Comment:
it's funny, they were named `interceptors` at the beginning, we changed to `matchingInterceptors` to be more expressive and now again to `interceptors` 😄

Reply:
I think it was because we had to distinguish it from the previously declared `interceptors` variable, but that variable has gone now.